### ISSUE TITLE
Rule `react-rules-of-hooks` false positive

### ISSUE THREAD

**Issue by jaydenseric** [denoland/deno_lint#1436]
### Lint Name

Rule `react-rules-of-hooks`

### Code Snippet

```ts
function A() {
  useA(a ?? "");
  useB();
  return null;
}
```

### Expected Result

No rule `react-rules-of-hooks` error.

### Actual Result

Rule `react-rules-of-hooks` error on the `useB()`:

```
Hook called conditionally
Move the hook call before you call any conditions. All hooks must be invoked on every render call
```

### Version

deno 2.3.1 (stable, release, aarch64-apple-darwin)
v8 13.5.212.10-rusty
typescript 5.8.3
